reflect/src/main/java/dagger/reflect/ReflectiveModuleParser.java

```diff
@@ -72,7 +84,12 @@ static void parse(Class<?> moduleClass, @Nullable Object instance,
 
         graphBuilder.add(key, binding);
       }
-      target = target.getSuperclass();
+
+      Class<?> superclass = target.getSuperclass();
+      if (superclass != Object.class && superclass != null) {
```

Comment:
add `Object.class` and `null` to initial value of `seen` and let it `continue;`?

Reply:
I'd prefer to keep these out of the collections altogether since they're more like bad data

Reply:
edge case: if the method is called with either of these, it can get inside the collections, and the error won't be graceful

Reply:
Sure but that's parameter validation which is a separate concern. If you pass null you get an NPE and if you pass `Object.class` you get an error that its methods don't have `@Provides`